airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py

```diff
@@ -10,24 +10,72 @@
 import pendulum
 from airbyte_cdk.sources.config import BaseConfig
 from facebook_business.adobjects.adsinsights import AdsInsights
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, Field, PositiveInt
 
 logger = logging.getLogger("airbyte")
 
 
 ValidFields = Enum("ValidEnums", AdsInsights.Field.__dict__)
 ValidBreakdowns = Enum("ValidBreakdowns", AdsInsights.Breakdowns.__dict__)
 ValidActionBreakdowns = Enum("ValidActionBreakdowns", AdsInsights.ActionBreakdowns.__dict__)
+DATE_TIME_PATTERN = "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$"
 
 
 class InsightConfig(BaseModel):
     """Config for custom insights"""
 
-    name: str = Field(description="The name value of insight")
-    fields: Optional[List[ValidFields]] = Field(description="A list of chosen fields for fields parameter", default=[])
-    breakdowns: Optional[List[ValidBreakdowns]] = Field(description="A list of chosen breakdowns for breakdowns", default=[])
+    class Config:
+        use_enum_values = True
+
+    name: str = Field(
+        title="Name",
+        description="The name value of insight",
+    )
+
+    fields: Optional[List[ValidFields]] = Field(
+        title="Fields",
+        description="A list of chosen fields for fields parameter",
+        default=[],
+    )
+
+    breakdowns: Optional[List[ValidBreakdowns]] = Field(
+        title="Breakdowns",
+        description="A list of chosen breakdowns for breakdowns",
+        default=[],
+    )
+
     action_breakdowns: Optional[List[ValidActionBreakdowns]] = Field(
-        description="A list of chosen action_breakdowns for action_breakdowns", default=[]
+        title="Action Breakdowns",
+        description="A list of chosen action_breakdowns for action_breakdowns",
+        default=[],
+    )
+
+    time_increment: Optional[PositiveInt] = Field(
+        title="Time Increment",
+        description=(
+            "Time window in days to get statistic from. The sync will chunk period between the start_date and the end_date by the number of days you specified. "
+            "For example if you specify time_increment = 7 then all statistics will be aggregated by 7 days starting from the start_date."
```

Comment:
```suggestion
            "For example if you set this value to 7, then all statistics will be reported as 7-day aggregates by starting from the start_date. If the start and end dates are October 1st and October 30th, then the connector will output 30 records, each of which represents aggregated stats from a 7 day lookback period."
```

Reply:
this is not true, we will have 5 periods:
- 01 - 06
- 07 - 13
- 14 - 20
- 21 - 27
- 28 - 30 ( 3 days)

Reply:
Ah. Can you just edit the comment suggestion to say that instead?